src/flows/recommendation_api/user_impressions_flow.py

```diff
@@ -1,49 +1,85 @@
-from prefect import task, Flow, Parameter, unmapped
-from prefect.tasks.aws.s3 import S3List
+from prefect import task, Flow, Parameter, context
+from api_clients.pocket_snowflake_query import PocketSnowflakeQuery
 import pandas as pd
+import datetime
 import boto3
 from sagemaker.feature_store.feature_group import FeatureGroup
 from sagemaker.session import Session
+from prefect.schedules import IntervalSchedule
 
+from utils import config
 from utils.flow import get_flow_name
 
 FLOW_NAME = get_flow_name(__file__)
 
-
-@task
-def extract_parquet_as_df(bucket:str, parquet_file: str):
-    parquet_file = f"s3://{bucket}/{parquet_file}"
-    return pd.read_parquet(parquet_file)
+BASE_QUERY = """
+WITH 
+prep as (
+  SELECT
+    a.HASHED_USER_ID,
+    a.CONTENT_ID,
+    c.RESOLVED_ID,
+    MAX(a.IMPRESSION_AGE) as max_impression_age,
+    SUM(a.IMPRESSION_COUNT) as total_imprs,
+    MAX(a.HAPPENED_AT_DAY) as HAPPENED_AT
+  FROM ANALYTICS.DBT.FIRST_IMPRESSED_AGE as a
+  JOIN ANALYTICS.DBT.CONTENT AS c
+    ON c.CONTENT_ID = a.CONTENT_ID
+  -- WHERE SLATE_LINEUP_ID in ('05027beb-0053-4020-8bdc-4da2fcc0cb68', '76b3f758-6aa3-430b-bcf1-8867897692b2')  -- wh persv2 lineups
+  --  AND CURRENT_DATE - HAPPENED_AT_DAY < 15  -- two weeks
+  WHERE CURRENT_DATE - DATE(a.TIME_ADDED) < 15  -- two weeks 
+  GROUP BY 1, 2, 3
+  )
+  
+SELECT
+    u.USER_ID,
+    p.HAPPENED_AT AS UPDATED_AT,
+    LISTAGG(DISTINCT p.RESOLVED_ID, ',') AS RESOLVED_IDS
+FROM prep as p
+JOIN ANALYTICS.DBT.USERS as u
+  ON p.HASHED_USER_ID = u.HASHED_USER_ID
+WHERE (p.max_impression_age > %(MAX_AGE)s) OR (p.total_imprs > %(MAX_IMPRS)s)
+GROUP BY 1, 2
+"""
```

Comment:
This query took a minute to execute. Might be fine in practice? If not, would be good to get G's eyes on it, given that he's our SQL expert.

Reply:
@gkatre can you have a look at the SQL in this flow?  I would like to run the flow daily and only update the feature store for `user_id`s that have new impression data